pyteal/ast/box.py

```diff
@@ -0,0 +1,226 @@
+from typing import TYPE_CHECKING
+from pyteal.ast.maybe import MaybeValue
+from pyteal.errors import TealInputError
+
+from pyteal.types import TealType, require_type
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class BoxCreate(Expr):
+    """Create a box with a given name and size."""
+
+    def __init__(self, name: Expr, size: Expr) -> None:
+        """
+        Args:
+            name: The key used to reference this box. Must evaluate to a bytes.
+            size: The number of bytes to reserve for this box. Must evaluate to a uint64.
+        """
+
+        super().__init__()
+        require_type(name, TealType.bytes)
+        require_type(size, TealType.uint64)
+        self.name = name
+        self.size = size
+
+    def __teal__(self, options: "CompileOptions"):
+        if options.version < Op.box_create.min_version:
+            raise TealInputError(
+                f"BoxCreate not available on teal version {options.version} (first available {Op.box_create.min_version})"
+            )
+
+        return TealBlock.FromOp(
+            options, TealOp(self, Op.box_create), self.size, self.name
+        )
+
+    def __str__(self):
+        return f"(box_create {self.name} {self.size})"
+
+    def type_of(self):
+        return TealType.none
+
+    def has_return(self):
+        return False
+
+
+BoxCreate.__module__ = "pyteal"
+
+
+class BoxDelete(Expr):
+    """Deletes a box given it's name"""
```

Comment:
@ahangsu Minor typo - Leaving _suggestion_ with what I think is intended.

```suggestion
    """Deletes a box given its name"""
```

Reply:
good find